src/Controller/Api/UserController.ts

```diff
@@ -19,7 +19,7 @@ export class UserController {
     public async registerUser(req: ValidatedRequest<UserRequestSchema>, res: Response) {
         try {
             const {email, password} = req.body;
-            const user = await this.userService.getUserByEmail(email);
+            const user = await this.userService.isAuthenticatedUser(email, password);
```

Comment:
This is now wrong logic. In this case you could have an existing user that's not logged in and have this result in false and allow user to register.